File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {

    let squareOfSum: Int
    let sumOfSquares: Int
    let differenceOfSquares: Int

    init(_ number: Int) {
        let sumOfNumbers = (number * (number + 1)) / 2
        self.squareOfSum = sumOfNumbers * sumOfNumbers
        self.sumOfSquares = (number * (number + 1) * (2 * number + 1)) / 6
        self.differenceOfSquares = self.squareOfSum - self.sumOfSquares
    }

}

File: r/difference-of-squares/difference-of-squares.R
difference_of_squares <- function(natural_number) { 

    sum_of_squares <- (natural_number * (natural_number + 1) * (2 * natural_number + 1)) / 6

    sum_of_numbers <- (natural_number * (natural_number + 1)) / 2
    square_of_sum = sum_of_numbers * sum_of_numbers

    return (square_of_sum - sum_of_squares)

}
